File: test-fixtures/csharp-service/CryptoService.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace CryptoService
{
    public class CryptoHelper
    {
        // --- AES encryption (FIPS Approved) ---
        public static byte[] EncryptAes(byte[] key, byte[] plaintext)
        {
            using var aes = Aes.Create();
            aes.Key = key;
            aes.GenerateIV();
            return aes.EncryptCbc(plaintext, aes.IV);
        }

        // --- AES-GCM (FIPS Approved) ---
        public static byte[] EncryptAesGcm(byte[] key, byte[] plaintext, byte[] nonce)
        {
            var ciphertext = new byte[plaintext.Length];
            var tag = new byte[16];
            using var gcm = new AesGcm(key, 16);
            gcm.Encrypt(nonce, plaintext, ciphertext, tag);
            return ciphertext;
        }

        // --- 3DES (FIPS Deprecated) ---
        public static byte[] EncryptTripleDes(byte[] key, byte[] plaintext)
        {
            using var tdes = TripleDES.Create();
            tdes.Key = key;
            return tdes.EncryptCbc(plaintext, tdes.IV);
        }

        // --- SHA-256 hash (FIPS Approved) ---
        public static byte[] HashSha256(byte[] data)
        {
            using var sha = SHA256.Create();
            return sha.ComputeHash(data);
        }

        // --- SHA-512 hash (FIPS Approved) ---
        public static byte[] HashSha512(byte[] data)
        {
            using var sha = SHA512.Create();
            return sha.ComputeHash(data);
        }

        // --- SHA-1 hash (FIPS Deprecated) ---
        public static byte[] HashSha1(byte[] data)
        {
            using var sha = SHA1.Create();
            return sha.ComputeHash(data);
        }

        // --- MD5 hash (NOT FIPS Approved) ---
        public static byte[] HashMd5(byte[] data)
        {
            using var md5 = MD5.Create();
            return md5.ComputeHash(data);
        }

        // --- HMAC-SHA256 (FIPS Approved) ---
        public static byte[] HmacSha256(byte[] key, byte[] data)
        {
            using var hmac = new HMACSHA256(key);
            return hmac.ComputeHash(data);
        }

        // --- RSA key generation (FIPS Approved) ---
        public static RSA GenerateRsaKey()
        {
            return RSA.Create(2048);
        }

        // --- ECDSA (FIPS Approved) ---
        public static ECDsa GenerateEcKey()
        {
            return ECDsa.Create(ECCurve.NamedCurves.nistP256);
        }

        // --- ECDH (FIPS Approved) ---
        public static ECDiffieHellman GenerateEcdhKey()
        {
            return ECDiffieHellman.Create(ECCurve.NamedCurves.nistP256);
        }

        // --- PBKDF2 (FIPS Approved) ---
        public static byte[] DeriveKeyPbkdf2(string password, byte[] salt)
        {
            using var pbkdf2 = new Rfc2898DeriveBytes(password, salt, 600000, HashAlgorithmName.SHA256);
            return pbkdf2.GetBytes(32);
        }
    }
}

File: test-fixtures/java-microservice/src/main/java/com/example/CryptoService.java
package com.example;

import javax.crypto.Cipher;
import javax.crypto.KeyAgreement;
import javax.crypto.Mac;
import javax.crypto.SecretKeyFactory;
import javax.crypto.spec.GCMParameterSpec;
import javax.crypto.spec.PBEKeySpec;
import javax.crypto.spec.SecretKeySpec;
import java.security.KeyPairGenerator;
import java.security.MessageDigest;
import java.security.SecureRandom;

/**
 * Example Java microservice demonstrating various crypto patterns.
 */
public class CryptoService {

    // ─── AES-GCM encryption (FIPS Approved) ─────────────────────────
    public byte[] encryptAesGcm(byte[] key, byte[] plaintext) throws Exception {
        Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
        byte[] iv = new byte[12];
        new SecureRandom().nextBytes(iv);
        GCMParameterSpec spec = new GCMParameterSpec(128, iv);
        cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, "AES"), spec);
        return cipher.doFinal(plaintext);
    }

    // ─── AES-ECB encryption (FIPS Deprecated) ───────────────────────
    public byte[] encryptAesEcb(byte[] key, byte[] plaintext) throws Exception {
        Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, "AES"));
        return cipher.doFinal(plaintext);
    }

    // ─── DESede / 3DES (FIPS Deprecated) ────────────────────────────
    public byte[] encryptTripleDes(byte[] key, byte[] plaintext) throws Exception {
        Cipher cipher = Cipher.getInstance("DESede/CBC/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, "DESede"));
        return cipher.doFinal(plaintext);
    }

    // ─── RSA key pair (FIPS Approved) ───────────────────────────────
    public void generateRsaKeyPair() throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
        kpg.initialize(2048);
        kpg.generateKeyPair();
    }

    // ─── ECDSA key pair (FIPS Approved) ─────────────────────────────
    public void generateEcKeyPair() throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC");
        kpg.initialize(256);
        kpg.generateKeyPair();
    }

    // ─── Ed25519 key pair (FIPS Approved) ───────────────────────────
    public void generateEd25519KeyPair() throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("Ed25519");
        kpg.generateKeyPair();
    }

    // ─── DH key agreement (FIPS Approved) ───────────────────────────
    public void performDhKeyAgreement() throws Exception {
        KeyAgreement ka = KeyAgreement.getInstance("DH");
    }

    // ─── X25519 key agreement (NOT FIPS Approved) ───────────────────
    public void performX25519KeyAgreement() throws Exception {
        KeyAgreement ka = KeyAgreement.getInstance("X25519");
    }

    // ─── SHA-256 hash (FIPS Approved) ───────────────────────────────
    public byte[] hashSha256(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        return md.digest(data);
    }

    // ─── SHA-1 hash (FIPS Deprecated) ───────────────────────────────
    public byte[] hashSha1(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("SHA-1");
        return md.digest(data);
    }

    // ─── MD5 hash (NOT FIPS Approved) ───────────────────────────────
    public byte[] hashMd5(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        return md.digest(data);
    }

    // ─── HMAC-SHA256 (FIPS Approved) ────────────────────────────────
    public byte[] hmacSha256(byte[] key, byte[] data) throws Exception {
        Mac mac = Mac.getInstance("HmacSHA256");
        mac.init(new SecretKeySpec(key, "HmacSHA256"));
        return mac.doFinal(data);
    }

    // ─── PBKDF2 (FIPS Approved) ─────────────────────────────────────
    public byte[] deriveKeyPbkdf2(char[] password, byte[] salt) throws Exception {
        SecretKeyFactory factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256");
        PBEKeySpec spec = new PBEKeySpec(password, salt, 600000, 256);
        return factory.generateSecret(spec).getEncoded();
    }
}
